docs/auto-instrumentation-install.md

```diff
@@ -0,0 +1,278 @@
+# Auto-instrumentation
+
+The process of adding observability code to your application so that it produces telemetry data is known as
+instrumentation. There are two main approaches to instrumentation in OpenTelemetry: auto-instrumentation and manual
+instrumentation.
+
+Auto-instrumentation
+
+- Usually is the easiest way to setup instrumentation for applications
+- OpentTelemetry leverages Kubernetes to implement auto-instrumentation in such a way that your applications can be
+  instrumented not by updating the application source code but rather by utilizing Kubernetes resources.
+- Supports a subset of instrumentation libraries compared to manual instrumentation
+
+Manual Instrumentation
+
+- Takes more setup effort compared to auto-instrumentation but offers more customizability
+- Requires editing application source code to include pre-built OpenTelemetry instrumentation libraries
+- Supports more instrumentation libraries and customization of the exported telemetry data compared to
+  auto-instrumentation
+
+In particular, auto-instrumentation is useful for applications that use widely popular frameworks and libraries, as
+these frameworks often have pre-built instrumentation capabilities already available.
+
+## Getting started with auto-instrumentation
+
+### 1. If the cert-manager available in the cluster, deploy it
+
+The cert-manager adds certificates and certificate issuers as resource types in Kubernetes clusters, and simplifies the
+process of obtaining, renewing and using those certificates. You can use the following make commands to deploy the
+cert-manager.
+
+```
+make cert-manager
```

Comment:
This implies that the user clones the repo first. Not sure this make target simplifies the user journey

Reply:
I've thought about this one, it is tricky. I included the necessary raw kubectl commands for this install as well to get around this for that reason.

We can't run a helm install with operator.enabmed=true unless the cert-manager is already up and running.

One solution I thought of but haven't had time to test is including the cert-manager as a subchart, I'm not sure what the order of operations would for all the subcharts installing thought. I'll try this solution today to see how plausible it is.

Reply:
Done